Add AdminUserDTO and the PublicUserResource

gzsombor opened the pull request:
To separate the publicly accessible user information from the private one.
 Porting https://github.com/jhipster/generator-jhipster/issues/12374

JasonTypesCodes commented:
Thanks @gzsombor !  It looks like the gradle-jwt-react sample end to end UI tests need to be checked with these updates.

atomfrede commented on generators/server/templates/src/main/java/package/service/dto/AdminUserDTO.java.ejs:
```diff
@@ -0,0 +1,185 @@
+        this.activated = user.getActivated();
+        this.imageUrl = user.getImageUrl();
+        this.langKey = user.getLangKey();
+        //this.createdBy = user.getCreatedBy();
```
Can this be removed?

gzsombor replied:
Yes, it was just copy-pasted from the original UserDTO

atomfrede reviewed (commented):
Looks good to me. Thanks @gzsombor for contributing this. @JasonTypesCodes We should now decide when to do the last/final release based on 6.x as this feature is already part of 7.x if I am not mistaken (and we will not backport it).

gzsombor commented:
I think the problem is that this blueprint still use the old - 6.x templates - for the frontend generation. 
I tried to migrate this blueprint to the latest generator, but I didn't have success.

JasonTypesCodes commented:
> Looks good to me. Thanks @gzsombor for contributing this. @JasonTypesCodes We should now decide when to do the last/final release based on 6.x as this feature is already part of 7.x if I am not mistaken (and we will not backport it).

I picked up the 1.0 conversation in its issue (#189)

gzsombor commented:
Sure, I will try to rebase my patch

mraible commented:
@gzsombor Can you please resolve conflicts or close?

gzsombor commented:
I've rebased the code, but I can't test it locally, using 'mhipster' or calling 'jhipster --blueprints micronaut' or 'npx jhipster --blueprints micronaut' ends up with this error:

```ERROR! ERROR! Named export 'isReservedTableName' not found. The requested module '../../../jdl/jhipster/reserved-keywords.js' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '../../../jdl/jhipster/reserved-keywords.js';
const { isReservedTableName } = pkg;

file:///mnt/butterfs/projects/jhipster/generator-jhipster/dist/generators/server/support/prepare-entity.mjs:21
import { isReservedTableName } from '../../../jdl/jhipster/reserved-keywords.js';
         ^^^^^^^^^^^^^^^^^^^
SyntaxError: Named export 'isReservedTableName' not found. The requested module '../../../jdl/jhipster/reserved-keywords.js' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '../../../jdl/jhipster/reserved-keywords.js';
const { isReservedTableName } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:190:5)
```

The developer setup is not documented at all, and I can't google it, how to solve. Any recommendation which works with the current JHipster generator?

mraible commented:
@gzsombor I just merged the [upgrade to JHipster 7.9.3](https://github.com/jhipster/generator-jhipster-micronaut/pull/282), so you might try rebasing again. I created an app last night with the following command after running `npm link`:

```
mhipster jdl ~/dev/generator-jhipster-micronaut/test-integration/samples/gradle-oauth-angular.jdl \
  --skip-jhipster-dependencies --force
```

gzsombor commented:
Thanks, it turned out, if I don't want to use the development jhipster-generator, it works properly :)

I've rebased, and fixed a couple of smaller issues around the DTO naming, I'm hopeful, that the tests will be green

mraible commented:
@gzsombor It looks like you need to run `npm run update-snapshot`.

gzsombor commented:
thanks, I've rebased the change, and fixed the silly search&replace typo bug... This will work now :)

gzsombor commented:
And everything is green now, we can merge it :)
